contracts/modules/staking/StakingPoolBase.sol

```diff
@@ -466,5 +473,37 @@ contract StakingPoolBase is Module, IERC900, CapperRole {
         emit Unstaked(personalStake.stakedFor, _amount, totalStakedFor(personalStake.stakedFor), _data);
     }
 
+    function withdrawStakes(bytes memory _data) internal returns (uint256){
+        StakeContract storage sc = stakeHolders[_msgSender()];
+        uint256 unstakeAmount = 0;
+        uint256 unstakedForOthers = 0;
+        uint256 personalStakeIndex = sc.personalStakeIndex;
+
+        uint256 i;
+        for (i = personalStakeIndex; i < sc.personalStakes.length; i++) {
+            Stake storage personalStake = sc.personalStakes[i];
+            if(personalStake.unlockedTimestamp > block.timestamp) break; //We've found last unlocked stake
+            
+            if(personalStake.stakedFor != _msgSender()){
+                //Handle unstake of staked for other address
+                stakeHolders[personalStake.stakedFor].totalStakedFor = stakeHolders[personalStake.stakedFor].totalStakedFor.sub(personalStake.actualAmount);
+                unstakedForOthers = unstakedForOthers.add(personalStake.actualAmount);
+                emit Unstaked(personalStake.stakedFor, personalStake.actualAmount, totalStakedFor(personalStake.stakedFor), _data);
+            }
+
+            unstakeAmount = unstakeAmount.add(personalStake.actualAmount);
+            personalStake.actualAmount = 0;
+        }
+        sc.personalStakeIndex = i;
+
+        uint256 unstakedForSender = unstakeAmount.sub(unstakedForOthers);
+        sc.totalStakedFor = sc.totalStakedFor.sub(unstakedForSender);
+        require(stakingToken.transfer(_msgSender(), unstakeAmount), "Unable to withdraw");
```

Comment:
There is no totalStakedFor reduction here.
Please add the line
totalStakedAmount = totalStakedAmount.sub(unstakeAmount);